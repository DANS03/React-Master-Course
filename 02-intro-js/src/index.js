//Ternary operator

const active = true;

/*let message =  '';

if( active ){
    message = 'Active'
}else{
    message = 'Inactive'
}  


All this code can be resume using a ternary operator
*/

const message = ( active ) ? 'Active' : 'Inactive';

console.log(message);


