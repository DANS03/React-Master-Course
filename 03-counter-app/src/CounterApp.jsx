import PropTypes from "prop-types";
import { useState } from "react";

export const CounterApp = ({ value }) => {
  const [counter, setCounter] = useState(value);

  const sum = () => {
    // we need to leave it outside due to we are changing the prop value
    //value = value + 1; //This will never be rerender but the value will be change
    setCounter(counter + 1);
    //console.log(counter);
  };

  const sus = () => {
    setCounter(counter - 1);
    //console.log(counter);
  };

  const res = () => {
    setCounter(value);
    //console.log(counter);
  };

  return (
    <>
      <h1>CounterApp</h1>
     
      <h2> {counter} </h2>

      <button onClick={(event) => sum(event)}> +1 </button>
      <button onClick={(event) =>  sus(event)}> -1 </button>
      <button onClick={(event) => res(event)}> Reset </button>
    </>
  );
};

CounterApp.propTypes = {
  value: PropTypes.number.isRequired,
};
