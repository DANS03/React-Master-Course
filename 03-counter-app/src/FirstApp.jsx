import PropTypes from 'prop-types';

const newMessage = "David"; //if a piece of code tha is never going to change is outside of the component this means that i will never be rerender when the component is

const newObj = {
  message: "Im an object",
  name: "objOne",
};

const newFunc = (name) => {
    return `Hello Im a function which receives a name, the name is: ${name}` 
};

export function App( { title , subtitle } ) 
{

  console.log(subtitle);

  //this is equivalent to this code : document.createElement....
  return (
    <>
      <h1>Hello World from React!</h1>
      <h2>Fragment component</h2>
      <p>{newMessage}</p>
      <p>{JSON.stringify(newObj)}</p>
      <p>{subtitle}</p>
      <strong>{newFunc(newMessage)}</strong>
      <h1 >This text comes form the props: === </h1><h1 data-testid='testTitle'>{title}</h1>
    </>
  );
}


App.propTypes = {
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.number
};  

App.defaultProps  = { //default props will always enter before the propTypes
  title : 'Im a default prop', 
  subtitle : 0,
  name: 'David' //You can create a prop here that can be use inside the component but will never be assing a  value
};