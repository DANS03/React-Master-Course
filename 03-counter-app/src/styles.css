html, body{ 
    background-color: #202221;
    color:white;
    font-family: 'Times New Roman', Times, serif;
    font-size: large;
    padding: 70px;
} 

button{
    padding: 15px;
    font-size: 1.3rem;
}