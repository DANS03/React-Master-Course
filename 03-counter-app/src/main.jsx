import React from 'react';
import  ReactDOM  from 'react-dom/client';
//import { App } from './FirstApp';
//import { HelloWorldApp } from './HelloWorldApp';
import { CounterApp } from './CounterApp';
import './styles.css'
//<App title="Hello I'm a prop!" subtitle={ 123 } ></App>
//<CounterApp value={ 123 }></CounterApp>

ReactDOM.createRoot( document.getElementById('root') ) .render(

    <React.StrictMode>
       <CounterApp value={ 123 }></CounterApp>
    </React.StrictMode>

);
