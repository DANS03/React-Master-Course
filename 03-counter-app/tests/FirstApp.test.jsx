import { render } from "@testing-library/react";
import { App } from "../src/FirstApp";

describe("TRest in First app component", () => {
  test("Must match the snapshot", () => {
    //This means that at the moment tha teh component is render it must match the OG snapshot

    const title = "Hello I'm a test";
    const { container } = render(<App title={title}></App>);
  
    expect(container).toMatchSnapshot();//the first time is render an snapshot will be crated in the folder with the same name
  });

  test("Must show the title", ()=>{
    const title = "Hello I'm a test";
    const { getByText, getByTestId } = render(<App title={title}></App>);

    expect( getByText("Hello I'm a test") ).toBeTruthy();

    expect ( getByTestId('testTitle').innerHTML ).toContain(title)

  });

  test('Must show the subtitle sended by props',() => {
    const subtitle = 123;
    const { getByText } = render(<App subtitle={subtitle}></App>);

    expect( getByText(subtitle) ).toBeTruthy();

  });



});
