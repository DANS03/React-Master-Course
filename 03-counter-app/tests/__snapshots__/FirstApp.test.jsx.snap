// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TRest in First app component Must match the snapshot 1`] = `
<div>
  <h1>
    Hello World from React!
  </h1>
  <h2>
    Fragment component
  </h2>
  <p>
    David
  </p>
  <p>
    {"message":"Im an object","name":"objOne"}
  </p>
  <p>
    0
  </p>
  <strong>
    Hello Im a function which receives a name, the name is: David
  </strong>
  <h1>
    This text comes form the props: === 
  </h1>
  <h1
    data-testid="testTitle"
  >
    Hello I'm a test
  </h1>
</div>
`;
